students/minchan/westagram/users/views.py

```diff
@@ -1,3 +1,93 @@
-from django.shortcuts import render
+import json
+import re
+from django.http.response   import JsonResponse
+from django.views           import View
+from users.models           import Users
+from django.http            import HttpResponse
 
 # Create your views here.
+class user_register(View) :
+    def valid_email(self,email) :
+        email_regular       = re.compile("[@][a-zA-Z]*[.]")
+        if email_regular.search(email) == None :
+            return False
+        return True
+
+
+    def is_email_duplicate(self,email) :
+        if Users.objects.filter(email=email) :
+            return True # 중복값이 있으면 True를 반환.
+        return False
+    
+
+    def valid_pw(self,pw) :
+        # 검사항목 - 0: 8자리 이상, 1:문자포함, 2:숫자포함, #:특수문자포함
+        test_case           = [0,0,0,0]
+        alphabet_regular    = re.compile("[a-zA-Z]")
+        number_regular      = re.compile("[\d]")
+        symbol_regular      = re.compile("[`~!@#$%^&*(),<.>/?]") #더 많은 특수문자의 허용은 [^0-9a-zA-Z]로 대체 가능.
+        #0 길이검사.
+        if len(pw) >= 8 : 
+            test_case[0] = 1
+
+        #1. 알파벳 포함 검사
+        for i in pw :
+            if alphabet_regular.match(i) != None :
+                test_case[1] = 1
+                break
+
+        #2. 숫자 포함 검사
+        for i in pw :
+            if number_regular.match(i) != None :
+                test_case[2] = 1
+                break
+
+        #3. 특수문자 포함 검사
+        for i in pw :
+            if symbol_regular.match(i) != None :
+                test_case[3] = 1
+                break
+
+        return test_case  
+
+
+    def post(self,request) :
+        data                = json.loads(request.body)
+        data_email          = data["email"]
+        data_pw             = data["password"]
```

Comment:
변수는 간결하고 명확하게 정해주셔야 합니다. data 용어는 제외해주세요.

```suggestion
        email    =  data["email"]
        password = data["password"]
```

Reply:
반영 했습니다.